src/System.Diagnostics.EventLog/src/System/Diagnostics/EventLogEntry.cs

```diff
@@ -0,0 +1,506 @@
+            {
+                eventKey = EventLog.GetEventLogRegKey(machineName, false);
+                if (eventKey == null)
+                    return null;
+                if (logName == null)
+                    logKey = eventKey.OpenSubKey("Application", /*writable*/false);
+                else
+                    logKey = eventKey.OpenSubKey(logName, /*writable*/false);
+                if (logKey == null)
+                    return null;
+                return logKey.OpenSubKey(source, /*writable*/false);
+            }
+            finally
+            {
+                if (eventKey != null)
+                    eventKey.Close();
+                if (logKey != null)
+                    logKey.Close();
+            }
+
+        }
+
+        private string GetMessageLibraryNames(string libRegKey)
+        {
+            // get the value stored in the registry
+            string fileName = null;
+            RegistryKey regKey = null;
+            try
+            {
+                regKey = GetSourceRegKey(owner.Log, Source, owner.MachineName);
+                if (regKey != null)
+                {
+                    fileName = (string)regKey.GetValue(libRegKey);
+                }
+            }
+            finally
+            {
+                if (regKey != null)
+                    regKey.Close();
+            }
+
+            if (fileName == null)
+                return null;
+
+            // convert any absolute paths on a remote machine to use the \\MACHINENAME\DRIVELETTER$ shares
+            if (owner.MachineName != ".")
+            {
+
+                string[] fileNames = fileName.Split(';');
+
+                StringBuilder result = new StringBuilder();
+
+                for (int i = 0; i < fileNames.Length; i++)
+                {
+                    if (fileNames[i].Length >= 2 && fileNames[i][1] == ':')
+                    {
+                        result.Append(@"\\");
+                        result.Append(owner.MachineName);
+                        result.Append(@"\");
+                        result.Append(fileNames[i][0]);
+                        result.Append("$");
+                        result.Append(fileNames[i], 2, fileNames[i].Length - 2);
+                        result.Append(';');
+                    }
+                }
+
+                if (result.Length == 0)
+                {
+                    return null;
+                }
+                else
+                {
+                    return result.ToString(0, result.Length - 1);
+                }
+            }
+            else
+            {
+                return fileName;
+            }
+        }
+
+
+        private short ShortFrom(byte[] buf, int offset)
+        {
+            // assumes little Endian byte order.
+            return (short)((0xFF00 & (buf[offset + 1] << 8)) | (0xFF & buf[offset]));
+        }
+
+        void ISerializable.GetObjectData(SerializationInfo info, StreamingContext context)
```

Comment:
we don't support serialization but keep the ISerialization interface implemented on types. That means you should remove the code inside the method here and throw a PNSE, e.g. like here: https://github.com/dotnet/corefx/pull/20220/files#diff-1e76675a85670323d082bc433cd0cfd6R628

Reply:
Same in `private EventLogEntry(SerializationInfo info, StreamingContext context)`. The body of that should be `throw new PlatformNotSupportedException`